File: fixtures/scale/scala/large/src/cryptobench/verify/RoundTrip.scala
package cryptobench.verify

import cryptobench.core.Cipher

object RoundTrip {
  def check(cipher: Cipher, sample: String): Boolean = {
    val encrypted = cipher.encrypt(sample)
    val decrypted = cipher.decrypt(encrypted)
    lenientMatches(sample, decrypted)
  }

  /** Compare on the cipher alphabet only: case, spacing and padding may differ. */
  private[verify] def lenientMatches(expected: String, actual: String): Boolean = {
    val left = expected.replaceAll("[^A-Za-z0-9]", "").toUpperCase
    val right = actual.replaceAll("[^A-Za-z0-9]", "").toUpperCase
    right.startsWith(left) || left.startsWith(right)
  }
}

File: fixtures/scale/java/large/src/cryptobench/verify/RoundTrip.java
package cryptobench.verify;

import cryptobench.core.Cipher;

public class RoundTrip {
    public static boolean check(Cipher cipher, String sample) {
        String encrypted = cipher.encrypt(sample);
        String decrypted = cipher.decrypt(encrypted);
        return matches(sample, decrypted);
    }

    /** Compare on the cipher alphabet only: case, spacing and padding may differ. */
    static boolean matches(String expected, String actual) {
        String left = expected.replaceAll("[^A-Za-z0-9]", "").toUpperCase();
        String right = actual.replaceAll("[^A-Za-z0-9]", "").toUpperCase();
        return right.startsWith(left) || left.startsWith(right);
    }
}
